File: algorithms/easy/cell_in_a_range_on_an_excel_sheet.ex
defmodule Solution do
  @spec cells_in_range(s :: String.t()) :: [String.t()]
  def cells_in_range(s) do
    [a, b] = String.split(s, ":")
    [r1, c1] = String.graphemes(a)
    [r2, c2] = String.graphemes(b)
    r1 = :binary.first(r1)
    r2 = :binary.first(r2)
    c1 = String.to_integer(c1)
    c2 = String.to_integer(c2)

    for i <- r1..r2, j <- c1..c2 do
      "#{List.to_string([i])}#{j}"
    end
  end
end

File: algorithms/easy/cell_in_a_range_on_an_excel_sheet.rs
impl Solution {
    pub fn cells_in_range(s: String) -> Vec<String> {
        let rc: Vec<&str> = s.split(":").collect();
        let a = rc[0].as_bytes();
        let b = rc[1].as_bytes();
        let r1 = a[0];
        let r2 = b[0];
        let c1 = a[1] - b'0';
        let c2 = b[1] - b'0';

        let mut ans = Vec::new();
        for i in r1..=r2 {
            for j in c1..=c2 {
                ans.push(format!("{}{}", i as char, j));
            }
        }
        ans
    }
}
